src/managed/Microsoft.NET.HostModel/AppHost/HostWriter.cs

```diff
@@ -21,6 +21,32 @@ public static class HostWriter
         private const string AppBinaryPathPlaceholder = "c3ab8ff13720e8ad9047dd39466b3c8974e592c2fa383d4a3960714caef0c4f2";
         private readonly static byte[] AppBinaryPathPlaceholderSearchValue = Encoding.UTF8.GetBytes(AppBinaryPathPlaceholder);
 
+        /// <summary>
+        /// The HostModel implements several services for updating the AppHost DLL.
+        /// These updates involve multiple file open/close operations.
+        /// An Antivirus scanner may intercept in-between and lock the file, 
+        /// causing the operations to fail with IO-Error.
+        /// So, the operations are retried a few times on IOException.
+        /// </summary>
+        /// <param name="func">The action to retry on IO-Error</param>
+        private static void RetryOnIOError(Action func)
+        {
+            uint numberOfRetries = 32;
```

Comment:
This might be too low - we would wait at the maximum 6.4 seconds. On spinning drives that's not that long. The extractor waits up to 50 seconds (and several people commented that it feels about right).

Reply:
I increased the iteration count to 256, the wait time is now 51.2s